probe-rs-target/src/flash_algorithm.rs

```diff
@@ -66,6 +66,9 @@ pub struct RawFlashAlgorithm {
     /// Address of the (non-standard) `ReadFlash()` entry point. Optional.
     #[serde(serialize_with = "hex_option")]
     pub pc_read: Option<u64>,
+    /// Address of the (non-standard) `BlankCheck()` entry point. Optional.
```

Comment:
`BlankCheck` is a standard CMSIS-PACK function. https://open-cmsis-pack.github.io/Open-CMSIS-Pack-Spec/main/html/algorithmFunc.html

Reply:
I had no idea. I'll update the comment.

Reply:
It's not just the comment. I'd prefer not breaking handwritten flash algos that decide to support "OurBlankCheck" when we decide to support the upstream function, so we should make sure we don't expect this custom addition to be called the same in the flash algo either.

Reply:
I can rename the function. Currently, `flash-algorithm` generates [a function called `BlankCheck`](https://github.com/probe-rs/flash-algorithm/commit/eb629a3d78b070e74226164461140b6cec0f1478), but I can submit a patch to update that as well.

What name would you like?

Reply:
I'm fine with whatever, even `BlankCheck2` 🫠 so long as it's different from the standard name.

Reply:
I'll name it `CustomBlankCheck` then, since it's for custom flash algorithms.

Reply:
I've renamed it, and [submitted a PR to flash-algorithm](https://github.com/probe-rs/flash-algorithm/pull/11) as well

Reply:
Why can this functionality that you implemented not just follow the standard? If I checked correctly, the standard also specifies a pattern to pass, which you don't atm, but nothing else is different. But there is not really anything keeping us from implementing it that way and then use the official callback. What am I overlooking?

Reply:
We could use that, but it's a big enough change that perhaps @bugadani didn't want that. I'm fine with either approach.

Reply:
I just thought we don't know the empty flash pattern upfront and can't use the standard method 🤔 I'd also prefer that, to be fair. The change itself should be largely the same - we just need to call the function with an extra argument.

Reply:
It doesn't say the pattern argument has to be used:

> The function BlankCheck can be used to check whether the specified block is empty, *or* whether the content is equal to a specific pattern defined in the argument **pat**.

I imagine that statement means that a function is free to ignore the third argument if it makes sense.

Let me rework the patch back to the original function name, and make the arguments line up with the CMSIS-DAP implementation.

Reply:
Algos carry the default erase value: https://open-cmsis-pack.github.io/Open-CMSIS-Pack-Spec/main/html/flashAlgorithm.html#AddFPA and we have it present in the target YAMLs :)

Reply:
Alright, `blank_check()` now has three arguments that follow the CMSIS-DAP implementation of `BlankCheck`